wgpu-core/src/device/resource.rs

```diff
@@ -147,8 +152,7 @@ impl Drop for Device {
     fn drop(&mut self) {
         resource_log!("Drop {}", self.error_ident());
 
-        let device_lost_closure = self.lock_life().device_lost_closure.take();
```

Comment:
thought: This change eliminates coverage of a `life_tracker` lock region over some cleanup operations after this hunk. I presume this is safe (from a place of ignorance), because these should be exclusively held resources. Is that right? I'm nervous about the `fence` in particular, because we acquire a `write` lock on it before `ManuallyDrop::take`ing it.

Reply:
As far as I understand in this case the lifetime of the guard didn't have to be extended, it's in a temporary scope.